### PR TITLE
PWN-7169 - koin module, manager holder

### PR DESCRIPTION
**Pull request by NikolaiEmelianov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7169

## Description of Work

koin module, manager holder, fixes

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/swap/SwapModule.kt [hunk 1]
```diff
@@ -134,5 +148,61 @@ object SwapModule : InjectionModule {
...
         factoryOf(::JupiterSwapInteractor)
+
+        factoryOf(::SwapStateRoutesRefresher)
+        factoryOf(::SwapWidgetMapper)
+        factoryOf(::SwapButtonMapper)
+
+        factory { (token: Token.Active?) ->
+            if (token == null) {
+                CommonSwapTokenSelector(
+                    get(),
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/SwapModule.kt:159]: named args

File: app/src/main/java/org/p2p/wallet/swap/SwapModule.kt [hunk 2]
```diff
@@ -134,5 +148,61 @@ object SwapModule : InjectionModule {
...
+                    get(),
+                    get(),
+                    token,
+                )
+            }
+        }
+        factory { (token: Token.Active?) ->
+            SwapStateInitialLoadingHandler(
+                get(),
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/SwapModule.kt:174]: named args

File: app/src/main/java/org/p2p/wallet/swap/SwapModule.kt [hunk 3]
```diff
@@ -134,5 +148,61 @@ object SwapModule : InjectionModule {
...
+                )
+            }
+            JupiterSwapPresenter(
+                widgetMapper = get(),
+                buttonMapper = get(),
+                stateManager = stateManager
+            )
+        } bind JupiterSwapContract.Presenter::class
     }
+    val swapStateManagerHolder = mutableMapOf<String, SwapStateManager>()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/SwapModule.kt:207]: put it on the top please

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/swap/SwapModule.kt:207]: and also extract holder into a class and inject it everywhere it needs to be 
so we dont need to work with DI layer in presenter

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateManagerHolder.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+object SwapStateManagerHolder {
+
+    private val swapStateManagerHolder = mutableMapOf<String, SwapStateManager>()
+
+    fun getOrCreate(key: String, defaultValue: () -> SwapStateManager): SwapStateManager =
+        swapStateManagerHolder.getOrPut(key) {
+            defaultValue()
+        }
+
+    fun get(key: String): SwapStateManager = swapStateManagerHolder[key]!!
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateManagerHolder.kt:12]: getValue can be used

File: app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateManagerHolder.kt [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package org.p2p.wallet.swap.jupiter.statemanager
+
+object SwapStateManagerHolder {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateManagerHolder.kt:3]: can be class, but singleOf in koin

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateManagerHolder.kt:3]: i do remind you that we are planning to write unit tests, and singletons via object are not preferred

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**